lib/rbac/filterer.rb

```diff
@@ -628,17 +628,34 @@ def scope_to_cloud_tenant(scope, user, miq_group)
     def scope_for_user_role_group(klass, scope, miq_group, user, managed_filters)
       user_or_group = miq_group || user
 
-      if user_or_group.try!(:self_service?) && MiqUserRole != klass
+      if user_or_group.try!(:self_service?) && klass != MiqUserRole
         scope.where(:id => klass == User ? user.id : miq_group.id)
       else
         role = user_or_group.miq_user_role
-        # hide creating admin group / roles from non-super administrators
+
+        # Exclude users/groups/roles tied to the super admin user if the current user isn't also
+        # a super admin user. This is to prevent creation / escalation to super admin.
         unless role&.super_admin_user?
-          scope = scope.with_roles_excluding(MiqProductFeature::SUPER_ADMIN_FEATURE)
+          # In the case that the user is not currently in a super admin group, but can _become_
+          #   a super admin by virtue of being in multiple groups, then they will be filtered by
+          #   with_roles_excluding. The allowed_ids option ensures the user can still see themselves.
+          #
+          # TODO: Determine if this same logic should apply to MiqUserRole as well
```

Comment:
So we would want role here to prevent a tenant admin from creating a group with super admin privs? i.e.: `"everything"`

Reply:
I'm not really sure here.  I need to allow the user themselves when it gets filtered out, but if a currently-non-admin user can be a super-admin, does that mean they should or shouldn't be able to see the super admin role?  I'm just not sure.

Reply:
the goal here is to prevent a tenant admin from creating a super user. (that they then know the password so they can escalate privileges.

You can do that via using a super group, or creating a group and assigning a super role